### PR TITLE
tplink-safeloader: Add support for TP-Link Deco M5

### PR DESCRIPTION
**Pull request by undisputed-seraphim**
Hello @oklona , I apologize for taking your changes and putting them under my name, but I would really like to move this project along. If you'd prefer that I don't do this please let me know. Otherwise, please review the changes I have made here. Again, I'm a little unsure about the block of code on L3684.

This change adds support for the TP-Link Deco-M5. I have checked IDs of the Deco-M5's sold here in Asia and they have the same ID numbers so it is not limited to EU models.

### CODE REVIEW

**Comment by oklona** [comment]
Hello. Yes, the L3684 - block was eliminated during the process with the first patch. -So that part is no longer relevant. It was replaced by a struct partition_names (L57 -> L102).

The default for partition_names is the normal style partition names, i.e. without the "@0, @1 postfixes, and each partition name can be overridden in the struct for the device itself. -So for the case of the M5, you need to add the following where line 1816 is in your patch:

```
.partition_names = {
			"partition_table@1",
			"soft_version@1",
			"os_image@1",
			"file-system@1",
			NULL
		}, 
```
-And obviously remove the L3684 block.

**Comment by undisputed-seraphim** [comment]
@oklona I have updated the pull request to match what you've said, please take a look.

If it's OK then we should contact one of the maintainers to review as well.

**Comment by undisputed-seraphim** [comment]
Edited the commit message to give a bit more context.

**Comment by oklona** [comment]
Yes, this now looks correct, and is exactly as I had planned it. The reason I didn't commit was that I experienced a few occurrences of the device not being able to find the root partition, so I had to investigate whether the changes to tplink-safeloader caused this. But in the end, I ended up seeing the exact same behavior even with stock firmware, so I do not believe tplink-safeloader is to blame :-)

**Comment by undisputed-seraphim** [comment]
Great! Let's try to get this merged then.

**Comment by undisputed-seraphim** [comment]
Not really sure who or where to notify for a review, perhaps @blocktrron ?

File: src/tplink-safeloader.c [hunk 1]
```diff
@@ -1742,6 +1742,87 @@ static struct device_info boards[] = {
...
+			{"file-system@1", 0x1190000, 0xc40000},
+			{"soft-version@1", 0x1dd0000, 0x10000},
+			{"profile@1", 0x1de0000, 0x10000},
+			{"default-config@1", 0x1df0000, 0x10000},
+			{"tm-sig", 0x1e00000, 0x200000},
+			{NULL, 0, 0}
+		},
+
+		.partition_names.partition_table = "partition-table@1",
+		.partition_names.soft_ver = "soft_version@1",
```
Comment (svanheule) [inline src/tplink-safeloader.c:1818]: I assume this underscore is a typo?

**Comment by svanheule** [comment]
Ideally, you should have an empty line after the title of your commit message (and no period at the end of the title). Github doesn't seem to mind this, but git tooling concatenates it into one long line.

While you're at it, a sorted `support_list` would also be nice.

**Comment by undisputed-seraphim** [comment]
> Ideally, you should have an empty line after the title of your commit message (and no period at the end of the title). Github doesn't seem to mind this, but git tooling concatenates it into one long line.
> 
> While you're at it, a sorted `support_list` would also be nice.

I didn't sort support_list because that is the list verbatim extracted from the official firmware image.

I've fixed other issues.

File: src/tplink-safeloader.c (hunk 1 above)
Comment (undisputed-seraphim) [reply src/tplink-safeloader.c]: Fixed.

**Comment by svanheule** [comment]
Thanks for the update! Patch applied to master.

**Pull request closed without merging**